Source/Core/Common/BitFieldView.h

```diff
@@ -0,0 +1,730 @@
+  constexpr bool operator==(const FixedViewArrayIterator& other) const { return idx == other.idx; }
+  constexpr bool operator!=(const FixedViewArrayIterator& other) const { return idx != other.idx; }
+};
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename field_t_, std::size_t start_, std::size_t width_, typename host_t_>
+class FixedViewArrayConstIterator
+{
+public:
+  using host_t = host_t_;
+  using field_t = field_t_;
+  using iterator_category = std::input_iterator_tag;
+  using value_type = field_t;
+  using difference_type = ptrdiff_t;
+  using pointer = void;
+  using reference = ConstBitFieldView<field_t, host_t>;
+  constexpr static std::size_t start = start_;
+  constexpr static std::size_t width = width_;
+
+private:
+  const host_t& host;
+  std::size_t idx;
+
+public:
+  // Required by std::input_or_output_iterator
+  constexpr FixedViewArrayConstIterator() = default;
+  // Required by LegacyIterator
+  constexpr FixedViewArrayConstIterator(const FixedViewArrayConstIterator& other) = default;
+  // Required by LegacyIterator
+  FixedViewArrayConstIterator& operator=(const FixedViewArrayConstIterator& other) = default;
+  // Move constructor and assignment operators, explicitly defined for completeness
+  constexpr FixedViewArrayConstIterator(FixedViewArrayConstIterator&& other) = default;
+  FixedViewArrayConstIterator& operator=(FixedViewArrayConstIterator&& other) = default;
+
+  FixedViewArrayConstIterator(const host_t& host_, const std::size_t idx_) : host(host_), idx(idx_)
+  {
+  }
+
+public:
+  FixedViewArrayConstIterator& operator++()
+  {
+    ++idx;
+    return *this;
+  }
+  FixedViewArrayConstIterator operator++(int)
+  {
+    FixedViewArrayConstIterator other(*this);
+    ++*this;
+    return other;
+  }
+  constexpr reference operator*() const { return reference(start + width * idx, width, host); }
+  constexpr bool operator==(const FixedViewArrayConstIterator other) const
+  {
+    return idx == other.idx;
+  }
+  constexpr bool operator!=(const FixedViewArrayConstIterator other) const
+  {
+    return idx != other.idx;
+  }
+};
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// BitField View (Array) in Member?  BitField View (Array) Macro?  BitField View (Array) Method?
+// You can believe whatever you want to believe this abbreviation is short for.
+
+#define BFVIEW_M(host, field_t, start, width, name)                                                \
+  constexpr BitFieldFixedView<field_t, start, width, decltype(host)> name()                        \
+  {                                                                                                \
+    return MakeBitFieldFixedView<field_t, start, width>(host);                                     \
+  }                                                                                                \
```

Comment:
There's a neat trick you can use here to automatically deduce the storage member for single-element structs:
```cpp
constexpr auto name() {
    auto [storage] = *this;
    return MakeBitFieldFixedView<field_t, start, width>(storage);
}
```
A possible trade-off here is that it requires you to use return type deduction, and of course you'd need a dedicated macro for structs with multiple members.

Reply:
That seems neat, but it doesn't seem worth it.  I know there are a good number of structs with multiple host members which will use BitFieldView once it gets in.

Reply:
It does remind me of an earlier version of the library, which had a split between macros for inheritance vs composition.  The one that worked via inheritance was much simpler to read the source code of...  I will consider it again after I can confirm the PRs dependent on this one are still functional. My main concern is hiding stuff behind meta-programming making the BFVIEW_M macro confusing.

Reply:
I just want to say this turned out to be an excellent suggestion.  I had my fears that auto type deduction would break something in one of the dependent PRs, but so far nothing too major has gone wrong.  Thank you for coming up with this!

Reply:
No problem, glad to hear it worked out well :)